pkg/report/linux.go

```diff
@@ -433,6 +438,183 @@ func symbolizeLine(symbFunc func(bin string, pc uint64) ([]symbolizer.Frame, err
 	return symbolized
 }
 
+type parsedOpcodes struct {
+	rawBytes       []byte
+	decompileFlags []DecompilerFlag
+	offset         int
+}
+
+func (ctx *linux) processOpcodes(codeSlice string) ([]string, error) {
+	parsed, err := ctx.parseLinuxOpcodes(codeSlice)
+	if err != nil {
+		return nil, err
+	}
+
+	// It is not guaranteed that the fragment of opcodes starts exactly at the boundary
+	// of a machine instruction. In order to simplify debugging process, we are trying
+	// to find the right starting position.
+	//
+	// We iterate over a fixed number of left boundaries. The exact number of iterations
+	// should strike a balance between the potential usefulness and the extra time needed
+	// to invoke the decompiler.
+	const opcodeAdjustmentLimit = 8
+
+	var currentBest []string
+
+	for leftCut := 0; leftCut < parsed.offset && leftCut < opcodeAdjustmentLimit; leftCut++ {
+		newBytes := parsed.rawBytes[leftCut:]
+		instructions, err := DecompileOpcodes(newBytes, parsed.decompileFlags, ctx.target)
+		if err != nil {
+			return nil, err
+		}
+
+		// We only want to return a response, where there exists a decoded instruction that
+		// perfectly aligns with the trapping instruction offset.
+		// At the same time, we'll do out best to find a code listing that does not contain
+		// unrecognized (bad) instuctions.
+
+		hasBad := false
+		for _, instruction := range instructions {
+			hasBad = hasBad || instruction.IsBad
```

Comment:
Hmm, interesting point.. I'll make it ignore (bad) at the last line - it is not informative anyway, since if all other instructions are parsed correctly and objdump's put an instruction boundary right at the offset we need, it is already the best possible decompilation.

Reply:
I think it's not necessary last, it can be last but one.

Reply:
Hmm, interesting point.. I'll make it ignore (bad) at the last line - it is not informative anyway, since if all other instructions are parsed correctly and objdump put an instruction boundary at the offset we need, it is already the best possible decompilation.

Reply:
So the safest approach would be to only check the instructions before the target offset

Reply:
Looks reasonable.